symfony/plugins/orangehrmTimePlugin/Api/TimeConfigPeriodAPI.php

```diff
@@ -0,0 +1,136 @@
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Time\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\ResourceEndpoint;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Service\MenuService;
+use OrangeHRM\Core\Traits\Service\ConfigServiceTrait;
+use OrangeHRM\Framework\Services;
+use OrangeHRM\Time\Api\Model\TimeConfigPeriodModel;
+use OrangeHRM\Time\Dto\TimeConfigPeriod;
+use OrangeHRM\Time\Service\TimeSheetPeriodService;
+
+class TimeConfigPeriodAPI extends Endpoint implements ResourceEndpoint
+{
+    use ConfigServiceTrait;
+
+    public const PARAMETER_START_DAY = 'startDay';
+
+    /**
+     * @var TimeSheetPeriodService|null
+     */
+    protected ?TimeSheetPeriodService $timeSheetPeriodService = null;
+
+    /**
+     * @return TimeSheetPeriodService
+     */
+    protected function getTimeSheetPeriodService(): TimeSheetPeriodService
+    {
+        if (!$this->timeSheetPeriodService instanceof TimeSheetPeriodService) {
+            $this->timeSheetPeriodService = new TimeSheetPeriodService();
+        }
+        return $this->timeSheetPeriodService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $status = $this->getTimeSheetPeriodService()->isTimesheetPeriodDefined();
+        $startDay = $status ? $this->getConfigService()->getTimeSheetPeriodConfig() : '2'; // to set monday as default
+        $timeConfigPeriod = new TimeConfigPeriod();
+        $timeConfigPeriod->setStartDay($startDay);
+        return new EndpointResourceResult(TimeConfigPeriodModel::class, $timeConfigPeriod);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID
+            ),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function update(): EndpointResult
+    {
+        $startDay = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_START_DAY);
+        $this->getTimeSheetPeriodService()->setTimesheetPeriod($startDay);
+        //After configuring the start day, we need to enable the menu items for time module
+        /** @var MenuService $menuService */
+        $menuService = $this->getContainer()->get(Services::MENU_SERVICE);
+        $menuService->enableModuleMenuItems('time');
+        $menuService->enableModuleMenuItems('attendance');
+        $menuService->enableModuleMenuItems('admin', ['Project Info', 'Customers', 'Projects']);
```

Comment:
thease menu items have to move under time module

Reply:
done